mava/utils/loggers/eval_json_logger.py

```diff
@@ -0,0 +1,153 @@
+        env_name: str,
+        task_name: str,
+        system_name: str,
+    ) -> None:
+        """Initialise JSON logger
+
+        Args:
+            experiment_path: path where experiment data should be logged
+            random_seed: random seed used for the experiment
+            env_name: name of environment of experiment. eg. "SMAC"
+            task_name: name of current experiment task. eg. "3m"
+            system_name: name of system being evaluated. eg. "IPPO"
+
+        Raises:
+            ValueError: if all the required parameters are not passed
+                in to the logger.
+        """
+
+        # Create subfolder for storing json file
+        self._log_dir = str(experiment_path) + f"/json_data/{env_name}/{task_name}/"
+        self._log_dir = paths.process_path(self._log_dir, add_uid=False)
+
+        self._logs_file_dir = (
+            f"{self._log_dir}{env_name}_{task_name}"
+            + f"_run{str(random_seed)}_evaluation_data.json"
+        )
+
+        self._step_count = 0
+        self._random_seed = str(random_seed)
+        self._env_name = env_name
+        self._task_name = task_name
+        self._system_name = system_name
+
+        # If directory doesn't exist create it
+        if not os.path.exists(self._log_dir):
+            os.makedirs(self._log_dir)
+
+        # Initialise json file for logging data
+        if not os.path.exists(self._logs_file_dir):
+            with open(self._logs_file_dir, "w+") as f:
+                json.dump(
+                    {
+                        self._env_name: {
+                            self._task_name: {
+                                self._system_name: {self._random_seed: {}}
+                            }
+                        }
+                    },
+                    f,
+                    indent=4,
+                )
+
+    def _jsonify_and_process(self, results_dict: Dict[str, Any]) -> None:
+        """Convert all elements to be logged to native python types."""
+
+        # convert all leaves to python types by calling tolist()
+        results_dict = jax.tree_util.tree_map(lambda leaf: leaf.tolist(), results_dict)
+
+        self._results_dict = results_dict
+
+    def _add_data_to_dictionary(
+        self, original_dictionary: Dict[str, Any], dictionary_to_add: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        """Adds new data to already logged data read in from a json file."""
+        if "step_count" not in list(dictionary_to_add.keys()):
+            original_dictionary[self._env_name][self._task_name][self._system_name][
+                self._random_seed
+            ]["absolute_metrics"] = dictionary_to_add
+        else:
+            original_dictionary[self._env_name][self._task_name][self._system_name][
+                self._random_seed
+            ][f"step_{str(self._step_count)}"] = dictionary_to_add
+        return original_dictionary
+
+    def write(self, results_dict: Dict[str, Any]) -> None:
```

Comment:
Do we know how this will behave with multiple evals writing or reading at once?  We can create an investigation if you think it is out of scope of this (for now, it might be worth doing a run to check it doesn't break, if you haven't already). 

I am not sure if we support concurrent eval right now?

Reply:
As discussed offline, we don't support multiple evaluators at the moment but it should be easy to extend this logger to be able to handle that.